xdsl/dialects/mpi.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+from xdsl.ir import OpResult, SSAValue, Dialect
+from xdsl.ir import ParametrizedAttribute
+
+from xdsl.irdl import (irdl_op_definition, Operation, Operand, Annotated,
+                       irdl_attr_definition, AnyAttr)
+
+
+@irdl_attr_definition
+class MPIRequest(ParametrizedAttribute):
+    name = "MPI_Request"
+
+
+@irdl_attr_definition
+class MPIStatus(ParametrizedAttribute):
+    name = "MPI_Status"
+
+
+@irdl_op_definition
+class MPIRequestOp(Operation):
+    # Name of the type. This is used for printing and parsing.
+    name: str = "mpi.MPIRequestOp"
+    request: Annotated[Operand, AnyAttr()]
+    # output: Annotated[OpResult, Attribute]
+
+    @staticmethod
+    def from_attr(request: AnyAttr) -> MPIRequestOp:
+        return MPIRequestOp.create(operands=[], result_types=[request])
+
+
+@irdl_op_definition
+class MPI_Wait(Operation):
```

Comment:
(looking at this [docu](https://www.mpich.org/static/docs/v3.1/www3/MPI_Wait.html)) This is another MPI thing that I don't quite understand. On the one hand, you pass it an MPI_Status (In my example above an MPI_STATUS_IGNORE). On the other hand, it seems to produce an MPI_Status, returning an int.

This nearly feels to me like it should have one more operand (an MPI_STATUS) that is an MPI_STATUS_IGNORE above. However, we might not want to model this just yet as we mainly use MPI_STATUS_IGNORE anyway.